### PR TITLE
Add preliminary CI

### LINKED ISSUE #78: Set up CI on the MultisigControl repo
In order to have the same process across engineering we need the usual CI set up on this repo:

As this is open source we should use GitHub Actions over Jenkins

- Compile
- Tests (run on every PR)
- Code-coverage results
- Linting (javascript and solidity)
- Clean up unused code/files (cc @C0deMunk33 to help/advise on this one)
- System test (investigate how easy/useful this would be)

What can be copied / leveraged from:
https://github.com/vegaprotocol/smart-contracts

### LINKED ISSUE #91: Audit javascript dependencies
It appears we are using a lot of dependencies, let's audit what's really necessary and keep only the minimal needed for this repository.

### PR DESCRIPTION
**Pull request by emilbayes**

### CODE REVIEW

File: test/Gather_Metrics.js [hunk 1]
```diff
@@ -206,7 +206,7 @@ contract("Gather Metrics", (accounts) => {
       console.log("Gas Results:")
 
       let signer_list = [accounts[0]]
-      for(let signer_count = 1; signer_count < accounts.length; signer_count++){
```
Comment (jeremyletang) [inline test/Gather_Metrics.js:209]: I see we create 50 accounts now, I guess we can keep this as is, 50 validators is our approx expectation at first when we open the validators set, so would be good to see the gas cost for up to this number.

Comment (emilbayes) [reply test/Gather_Metrics.js:209]: Ah yeah, I just added 50 for good measure, but then this test expected the number of accounts to be equal to the 20 private keys it generates at the top

**Review by jeremyletang — DISMISSED** [review]

File: test/Gather_Metrics.js (hunk 1 above)
Comment (jeremyletang) [reply test/Gather_Metrics.js:209]: indeed, but if we can make it up to 50 without too much trouble, then that would be very valuable info.

Comment (emilbayes) [reply test/Gather_Metrics.js:209]: Let me do that

Comment (jeremyletang) [reply test/Gather_Metrics.js:209]: cheers

**Review by jeremyletang — APPROVED** [review]

**Review by najienka — APPROVED** [review]

**Pull request merged**